redash/models.py

```diff
@@ -1000,6 +1006,32 @@ def outdated_queries(cls):
 
         return outdated_queries.values()
 
+    @classmethod
+    def delete_stale_resultsets(cls):
+        delete_count = 0
+        queries = Query.query.filter(Query.schedule_resultset_size != None).order_by(Query.schedule_resultset_size.desc())
+        # Multiple queries with the same text may request multiple result sets
+        # be kept. We start with the one that keeps the most, and delete both
+        # the unneeded bridge rows and result sets.
+        first_query = queries.first()
+        if first_query is not None and queries[0].schedule_resultset_size:
```

Comment:
New to sqlalchemy here, how is `queries.first()` different from `queries[0]`?

Reply:
[http://docs.sqlalchemy.org/en/rel_1_1/orm/query.html#sqlalchemy.orm.query.Query.first](Return the first result of this Query or None if the result doesn’t contain any row.)

You're right that it's a bit odd to then use `queries[0]` here; fixed.